cpp/ql/test/library-tests/ir/ir/raw_ir.expected

```diff
@@ -7433,6 +7441,309 @@ ir.cpp:
+# 1371|     mu1371_9(unknown)        = ^CallSideEffect                   : ~m?
+# 1371|     mu1371_10(String)        = ^IndirectMayWriteSideEffect[-1]   : &:r1371_2
+# 1371|     v1371_11(void)           = ^BufferReadSideEffect[0]          : &:r1371_7, ~m?
+# 1371|     mu1371_12(unknown)       = ^BufferMayWriteSideEffect[0]      : &:r1371_7
+# 1371|     r1371_13(String)         = Load[#temp1371:17]                : &:r1371_2, ~m?
+# 1371|     v1371_14(void)           = Call[acceptValue]                 : func:r1371_1, 0:r1371_13
+# 1371|     mu1371_15(unknown)       = ^CallSideEffect                   : ~m?
+# 1372|     r1372_1(glval<unknown>)  = FunctionAddress[acceptValue]      : 
+# 1372|     r1372_2(glval<String>)   = VariableAddress[#temp1372:25]     : 
+# 1372|     mu1372_3(String)         = Uninitialized[#temp1372:25]       : &:r1372_2
+# 1372|     r1372_4(glval<unknown>)  = FunctionAddress[String]           : 
+# 1372|     r1372_5(glval<char[4]>)  = StringConstant["foo"]             : 
+# 1372|     r1372_6(char *)          = Convert                           : r1372_5
+# 1372|     v1372_7(void)            = Call[String]                      : func:r1372_4, this:r1372_2, 0:r1372_6
+# 1372|     mu1372_8(unknown)        = ^CallSideEffect                   : ~m?
+# 1372|     mu1372_9(String)         = ^IndirectMayWriteSideEffect[-1]   : &:r1372_2
+# 1372|     v1372_10(void)           = ^BufferReadSideEffect[0]          : &:r1372_6, ~m?
+# 1372|     mu1372_11(unknown)       = ^BufferMayWriteSideEffect[0]      : &:r1372_6
+# 1372|     r1372_12(String)         = Load[#temp1372:25]                : &:r1372_2, ~m?
+# 1372|     v1372_13(void)           = Call[acceptValue]                 : func:r1372_1, 0:r1372_12
+# 1372|     mu1372_14(unknown)       = ^CallSideEffect                   : ~m?
+# 1373|     r1373_1(glval<String>)   = VariableAddress[#temp1373:5]      : 
+# 1373|     mu1373_2(String)         = Uninitialized[#temp1373:5]        : &:r1373_1
+# 1373|     r1373_3(glval<unknown>)  = FunctionAddress[String]           : 
+# 1373|     v1373_4(void)            = Call[String]                      : func:r1373_3, this:r1373_1
+# 1373|     mu1373_5(unknown)        = ^CallSideEffect                   : ~m?
+# 1373|     mu1373_6(String)         = ^IndirectMayWriteSideEffect[-1]   : &:r1373_1
+# 1373|     r1373_7(String)          = Load[#temp1373:5]                 : &:r1373_1, ~m?
+# 1373|     r1373_8(String)          = Convert                           : r1373_7
+# 1373|     r1373_9(glval<unknown>)  = FunctionAddress[c_str]            : 
+# 1373|     r1373_10(char *)         = Call[c_str]                       : func:r1373_9, this:r1373_8
+# 1373|     mu1373_11(unknown)       = ^CallSideEffect                   : ~m?
+# 1373|     v1373_12(void)           = ^BufferReadSideEffect[-1]         : &:r1373_8, ~m?
+# 1373|     mu1373_13(String)        = ^IndirectMayWriteSideEffect[-1]   : &:r1373_8
```

Comment:
Shouldn't the `this` operand to a `CallInstruction` have a pointer, reference, or `glval<...>` type? I think it also breaks the `AddressOperand`s for the side effects.

There's the same issue with the method calls on lines 1374, 1385, 1386, 1397 and 1398.

Reply:
These should all be fixed now, with a consistency check added.